### ISSUE TITLE
shape mismatch in DC_and_CE_smooth_loss - cross_entropy_loss

### ISSUE THREAD

**Issue by Grizemc** [zhiqin1998/UMamba2#3]
<img width="988" height="1350" alt="Image" src="https://github.com/user-attachments/assets/b5be5ea4-c8a6-4688-ab71-8cf4acab23e0" />

**Comment by zhiqin1998** [comment]
> Thank you very much. The printed result is shown in the picture:<img alt="Image" width="1279" height="957" src="https://private-user-images.githubusercontent.com/152372912/512773850-bfc15f0f-b91e-43ad-aa8e-cb1ee6071ae6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI4NzM1MTUsIm5iZiI6MTc2Mjg3MzIxNSwicGF0aCI6Ii8xNTIzNzI5MTIvNTEyNzczODUwLWJmYzE1ZjBmLWI5MWUtNDNhZC1hYThlLWNiMWVlNjA3MWFlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTExMVQxNTAwMTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOWIxZGE0Y2FmNWFlMTcyY2E4ZDMzNTRlN2FiM2NjYTAwMzQ1MzU5ZWYwMzk5YjMzMTQ2ZDY4MWYzYzY3MzI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ysYpuLl1btr4fhML74_lw2-tASCxa_sAIqU7bnjxdBw">

Please check the shape of `net_output` which is the predicted logits.

> I'm a little curious why using the nnUNetTrainer() class jumps to nnUNetTrainer_TF3_Task1. Where is it specified?

This should never happen because `nnUNetTrainer_TF3_Task1` inherits the `nnUNetTrainer` class, not the other way around. The trainer class is specified when you run training with `nnUNetv2_train`. e.g. the `-tr` argument below.
```
nnUNetv2_train {dataset_id} 3d_fullres_torchres_mambabot2_ps160x288x288_bs1 all -p nnUNetResEncUNetLPlans_torchres -tr **nnUNetTrainer_TF3_Task1_1500ep_accum2** -pretrained_weights <path to your pretarined weights>
```

**Comment by zhiqin1998** [comment]
Yes, make sure the dataset.json in `nnUNet_raw` and `nnUNet_preprocessed` only has a max class ID of 46 (i.e. replace them with the correct `mapped_dataset.json`)

**Comment by Grizemc** [comment]
> 是的，请确保 dataset.json 文件中的`nnUNet_raw`类`nnUNet_preprocessed`ID 最大值为 46（即替换为正确的值`mapped_dataset.json`）。

thankyou very much！！！！ but i have a problem，
How long does it take for you to train once？Because I feel like I've been training for too long


<img width="1333" height="734" alt="Image" src="https://github.com/user-attachments/assets/48f3bd21-7832-426e-b2ce-ef3f3ec883f7" />

**Comment by zhiqin1998** [comment]
Great to know you got it working. My per epoch time is around 4 minutes, so roughly 3 days for 1000 epochs on a RTX4090 GPU.

My advice is try to maximize GPU time, e.g. save all data in an SSD to minimize data loading time, set `export nnUNet_n_proc_DA={number_of_cpus}` before training to utilize all CPUs
